### PR TITLE
fix(ingest/bq): do not query PARTITIONS for biglake tables

### PR DESCRIPTION
**Pull request by mayurinehate**
- allows ingestion of biglake tables when profiling is enabled.
- include dataset location as properties
- mark external tables with subtype
- honour profiling.profile_external_tables config for bigquery

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/common/subtypes.py [hunk 1]
```diff
@@ -17,6 +17,8 @@ class DatasetSubTypes(StrEnum):
     POWERBI_DATASET_TABLE = "PowerBI Dataset Table"
     QLIK_DATASET = "Qlik Dataset"
     BIGQUERY_TABLE_SNAPSHOT = "Bigquery Table Snapshot"
+    SHARDED_TABLE = "sharded table"
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/common/subtypes.py:20]: should this be capitalized?

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py [hunk 2]
```diff
@@ -252,7 +253,16 @@ def get_datasets_for_project_id(
             self.report.num_list_datasets_api_requests += 1
             datasets = self.bq_client.list_datasets(project_id, max_results=maxResults)
             return [
-                BigqueryDataset(name=d.dataset_id, labels=d.labels) for d in datasets
+                BigqueryDataset(
+                    name=d.dataset_id,
+                    labels=d.labels,
+                    location=(
+                        d._properties["location"]
```
[the remark this answers is not in the document]
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py:260]: As per docs, yes, it should always have location. I'll make the change. Looks safer.

File: metadata-ingestion/src/datahub/ingestion/source/common/subtypes.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/common/subtypes.py:20]: Ideally yes. Kept as is in favour of earlier usage. Not sure, if UI handles this case change well. for example - filtering by subtype could show two entries `"sharded table"` and `"Sharded Table"`. Doesn't seem so harmful to make the change. Will change this.

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/common/subtypes.py]: image

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/common/subtypes.py]: This should be temporary behavior as eventually all sharded tables would get same subtype.

**Pull request merged**